### ISSUE TITLE
FastBoot is broken

### ISSUE THREAD

**Issue by rmmmp** [firebase/emberfire#500]
So the FastBoot for EmberFire is broken. When fetching data, the promise doesn't resolve or catch an error.

In my investigation, I tried using Firebase Admin SDK for node.js instead of Firebase Web SDK in EmberFire and it works. So I'm led to believe that the web SDK no longer works in node.js environment, which is weird since that's not what it says [here](https://firebase.google.com/docs/database/admin/start#admin-sdk-setup).

Can someone reconfirm this bug in their apps? Should we be integrating Firebase Admin SDK for FastBoot environment? My problem with integrating the Admin SDK is that there needs to be a standard way to pass in the secret keys to FastBoot without it existing in `config/environment.js` so we don't risk exposing it to the user.

EDIT:

Enabling logging shows the following:

>p:0: Making a connection attempt
>getToken() completed. Creating connection.
>c:0:23: Connection created
>p:0: Failed to get token: Error: No transports available
>p:0: data client disconnected

There's someone in SO that's having the same [issue](https://stackoverflow.com/questions/43372742/cant-get-firebase-in-node-js-app-to-run) of running the web SDK in node.js.

EDIT AGAIN:

Here's my current setup.

```javascript
DEBUG: -------------------------------
DEBUG: Ember      : 2.10.2
DEBUG: Ember Data : 2.10.0
DEBUG: Firebase   : 3.8.0
DEBUG: EmberFire  : 2.0.6
DEBUG: jQuery     : 3.2.1
DEBUG: -------------------------------
```

After trying it with EmberFire 2.0.5, FastBoot works. I'll try using EmberFire 2.0.6 with the latest Ember Data and post back here what I find.

**Comment by rmmmp** [comment]
Looks like using the latest Ember Data doesn't fix things. I'm unable to determine what's in 2.0.6 that's causing the FastBoot bug.

**Comment by PrinceCornNM** [comment]
This has been a problem for us since 2.0.6 first came out, I think it might be related to the node version of firebase not playing well with the updates made to emberfire in 2.0.6.

**Comment by rmmmp** [comment]
@PrinceCornNM, for what it's worth, I tried reverting all changes to 2.0.5 except for the moving of Firebase from Bower to NPM and it still didn't work. I'll try setting up a 2.0.6 with Firebase on Bower and see what happens though it'll be a bit odd if that's the reason why FastBoot isn't working.

**Comment by rmmmp** [comment]
So I've had some spare time to look into this today and I've figured out the problem. The hunch is right and it's related to the recent change of moving the Firebase dependency from bower to npm.

FastBoot dependencies can be used by using the `FastBoot.require()` method. Since we removed the shim, in 2.0.6, the AMD module used in FastBoot isn't coming from `FastBoot.require()`. We need to add the following shim to make it work:

```javascript
// vendor/shims/firebase.js
(function() {
  function vendorModule() {
    'use strict';

    return { 'default': FastBoot.require('firebase') };
  }

  if (window.FastBoot) {
    define('firebase', [], vendorModule);
  }
})();
```

I'll try to make a PR later.
